Fix frontend build

agpituk opened the pull request:
# What's changing

We found out currently our frontend bulid wasn't working properly. The root cause is that the .env file wasn't being passed properly to the npm command, which meant the VUE_APP_BASE_URL wasn't used inside the Dockerfile.

In order to fixed that I've modified several files to ensure consistency accross the project. One of the key things is that the .env.example file has changed to avoid inline comments , which means everyone will need to ensure their .env file is aligned with that once this is merged.


# How to test it

Steps to test the changes:

1. Spin up the Frontend locally (Run make local-up or start-lumigator-build)
2. Check that the frontend can connect to the backend)



# I already...

- [X] Tested the changes in a working environment to ensure they work as expected
- [N/A] Added some tests for any new functionality
- [N/A ] Updated the documentation (both comments in code and [product documentation](https://mozilla-ai.github.io/lumigator) under `/docs`)
- [X] Checked if a (backend) DB migration step was required and included it if required

ividal commented:
If it's any help, I've been testing this without the .env under frontend and launching an experiment, seeing results and downloading all works as expected (`local-up` and `start-lumigator`).

I was wondering why the regex was necessary - but will wait until you're done with this, @agpituk

agpituk commented:
> I was wondering why the regex was necessary - but will wait until you're done with this, @agpituk
Thanks for that @ividal . I definitely think we should improve this in a follow up, probably using [dotenv](https://www.npmjs.com/package/dotenv) or something like that. The regex is definitely something I don't like (if you mess with the `.env` it won't pull the envs properly for the frontend). Should we merge it as it is and have a follow up after MVP?

agpituk commented:
> > I was wondering why the regex was necessary - but will wait until you're done with this, @agpituk
> 
> Thanks for that @ividal . I definitely think we should improve this in a follow up, probably using [dotenv](https://www.npmjs.com/package/dotenv) or something like that. The regex is definitely something I don't like (if you mess with the `.env` it won't pull the envs properly for the frontend). Should we merge it as it is and have a follow up after MVP?


I actually wasn't happy with this after reading it and I decided to do another approach. After reading [vite doc ](https://v2.vitejs.dev/config/#async-config)I just removed the regex and changed vite configuration to point to our env file. Please have another look into it @ividal @george-mzai @macaab26

macaab26 commented:
@agpituk Reviewed, looks and works great 😄
